models/user_pinnedrepo.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/modules/structs"
+
+	"xorm.io/builder"
+)
+
+// UserPinnedRepo represents a pinned repo by an user or org.
+type UserPinnedRepo struct {
+	ID      int64 `xorm:"pk autoincr"`
+	UID     int64 `xorm:"INDEX NOT NULL"`
+	RepoID  int64 `xorm:"INDEX NOT NULL"`
+	IsOwned bool  `xorm:"INDEX NOT NULL"`
```

Comment:
If the repository is transfered to another user?

Reply:
Sorry, I forgot.